src/main/java/athena/Parser.java

```diff
@@ -0,0 +1,261 @@
+package athena;
+
+/**
+ * Handles parsing of user input
+ */
+public class Parser {
+
+    public static final String COMMAND_WORD_DELIMITER = " ";
+    public static final String NAME_DELIMITER = "n/";
+    public static final String TIME_DELIMITER = "t/";
+    public static final String DURATION_DELIMITER = "d/";
+    public static final String DEADLINE_DELIMITER = "D/";
+    public static final String RECURRENCE_DELIMITER = "r/";
+    public static final String IMPORTANCE_DELIMITER = "i/";
+    public static final String ADDITIONAL_NOTES_DELIMITER = "a/";
+    /**
+     * Parses user input and recognises what type of command
+     * and parameters the user typed
+     * @param input String representing user input
+     * @return new Command object based on what the user input is
+     */
+    public static Command parse(String input) {
+        String[] commandAndDetails = input.split(COMMAND_WORD_DELIMITER, 2);
+        String commandType = commandAndDetails[0];
+        String taskInfo = commandAndDetails[1];
+        Command command = null;
+
+        int nPosition = taskInfo.indexOf(NAME_DELIMITER);
+        int tPosition = taskInfo.indexOf(TIME_DELIMITER);
+        int dPosition = taskInfo.indexOf(DURATION_DELIMITER);
+        int DPosition = taskInfo.indexOf(DEADLINE_DELIMITER);
+        int rPosition = taskInfo.indexOf(RECURRENCE_DELIMITER);
+        int iPosition = taskInfo.indexOf(IMPORTANCE_DELIMITER);
+        int aPosition = taskInfo.indexOf(ADDITIONAL_NOTES_DELIMITER);
+
+        if (commandType.equals("add")) {
+            String[] retrieveName = taskInfo.split(NAME_DELIMITER, 2);
+            String retrievedName = retrieveName[1];
+            String name;
+            int nameNextSlash = retrievedName.indexOf("/");
+            if (nameNextSlash == -1) {
+                name = retrievedName;
+            } else {
+                name = retrievedName.substring(0, (nameNextSlash - 2));
+            }
```

Comment:
We should refactor this part out since it is repeated like 10 times, but I think it is ok to do it after tomorrow's meeting.